### PR TITLE
DOCS-2113: Document prerelease versioning

### PR DESCRIPTION
**Pull request by JessamyT**

### CODE REVIEW

File: docs/registry/advanced/iterative-development.md [hunk 1]
```diff
@@ -278,3 +279,20 @@ If you are prompted with a message similar to `Enter passphrase for key '/Users/
...
+## Use prerelease versioning
+
+To publish a module version that is not yet fully tested, you can publish it as a prerelease (also called "release candidate") version of a module to the modular registry.
+Publishing a prerelease version will not affect any machines that are using the existing module.
+Regardless of the machine’s version pinning setting, only machines that are set to the exact prerelease version will be updated.
+
+For example, imagine your latest stable version is `0.1.2`.
+If you publish a prerelease version tagged `0.1.2-rc0`, all machines continue to use version `0.1.2`.
+Similarly, if you publish a prerelease version tagged `0.1.3-rc0`, all machines will stay on version `0.1.2`.
+If you pin a machine to the exact version number `0.1.3-rc0`, only then will that machine use the prerelease version.
```
Comment (JessamyT) [inline docs/registry/advanced/iterative-development.md:292]:
```suggestion
If you pin a machine to the exact version number `0.1.3-rc0`, only then will that machine use the prerelease version.

{{% alert title="Note" color="note" %}}
If you are using cloud build, be sure to update your [GitHub action file](/registry/upload/#update-an-existing-module-using-a-github-action) tags to include the release candidate version.
You can put `"*"` to trigger the build action on all tags regardless of correct syntax, or you can use a [regex](https://regex101.com/r/Ly7O1x/3/) to trigger the build action on all tags that have valid syntax.
{{% /alert %}}
```
@michaellee1019 is this a good way to put this?

Comment (michaellee1019) [reply docs/registry/advanced/iterative-development.md:292]: Overall looks good. As discussed lets put the whole semver 2.0 regex here as a code block with copy button. Thank you!

File: docs/registry/advanced/iterative-development.md [hunk 2]
```diff
@@ -278,3 +279,31 @@ If you are prompted with a message similar to `Enter passphrase for key '/Users/
...
+
+Your tags should adhere to [semantic versioning specification (SemVer)](https://semver.org), meaning that the tag should begin with a major version number, then a minor version number, then a patch version number, separated by periods.
+It is up to you as the developer to choose when to increment each number.
+You can append any label after the major, minor, and patch version numbers.
+For example, you can label your prerelease version `0.1.2-rc0` or `0.1.2-beta`.
+If your tag does not adhere to SemVer, cloud builds will fail.
+
+### Update your GitHub action file
+
+If you are using cloud build, be sure to update your [GitHub action file](/registry/upload/#update-an-existing-module-using-a-github-action) tags to include the release candidate version.
```
Comment (sguequierre) [inline docs/registry/advanced/iterative-development.md:304]:
```suggestion
If you are using Cloud Build, be sure to update your [GitHub action file](/registry/upload/#update-an-existing-module-using-a-github-action) tags to include the release candidate version.
```
Or Cloud build-- i think, just to match the product name?

**Review by sguequierre — APPROVED** [review]
looks excellent!

**Pull request merged**